collector/binlog.go

```diff
@@ -72,7 +72,7 @@ func (ScrapeBinlogSize) Version() float64 {
 }
 
 // Scrape collects data from database connection and sends it over channel as prometheus metric.
-func (ScrapeBinlogSize) Scrape(ctx context.Context, db *sql.DB, ch chan<- prometheus.Metric, logger log.Logger) error {
+func (ScrapeBinlogSize) Scrape(ctx context.Context, db *sql.DB, ch chan<- prometheus.Metric, logger log.Logger, semanticVersionIsNewer bool) error {
```

Comment:
This `semanticVersionIsNewer` is not a good way to handle this. It's too specific to one collector, and hard-codes it as a flag for every collector.

What we probably want is to do something like we did in the postgres_expoter, which is to make the `db` instance include both a database connection and a semver that each collector can access.

Reply:
Do you have a link to the related code?

Reply:
Here's the PR: https://github.com/prometheus-community/postgres_exporter/pull/785

Reply:
I would prefer to do that refactoring as a separate PR.

Reply:
I will look into this, but it will take some time...

Reply:
I created a PR with that refactoring https://github.com/prometheus/mysqld_exporter/pull/859